### PR TITLE
Fix: auto figures and caption

### PR DESCRIPTION
**Pull request by saugatmalla**
Revamp auto figure and caption code 
- fix the figures that don't display without captions
- fix the figures with picture and caption in the same `p` element
Test URLs: 
- Before: https://main--blog--adobecom.hlx.page/drafts/rclayton/a-fresh-look-at-data-elements
- After: https://figure--blog--adobecom.hlx.page/drafts/rclayton/a-fresh-look-at-data-elements

### CODE REVIEW

File: scripts/utils.js [hunk 1]
```diff
@@ -43,30 +43,50 @@ export const [setLibs, getLibs] = (() => {
...
-      loadStyle(`${miloLibs}/blocks/figure/figure.css`);
+  const imgEls = document.querySelectorAll('main > div > p > picture');
+  if (!imgEls.length) return;
+
+  const { createTag, loadStyle } = await import(`${miloLibs}/utils/utils.js`);
+  loadStyle(`${miloLibs}/blocks/figure/figure.css`);
+
+  imgEls.forEach((imgEl) => {
+    const block = createTag('div', { class: 'figure' });
+    const row = createTag('div', null);
```
Comment (meganthecoder) [inline scripts/utils.js:72]:
```suggestion
    const row = createTag('div');
```

File: scripts/utils.js [hunk 2]
```diff
@@ -43,30 +43,50 @@ export const [setLibs, getLibs] = (() => {
...
+    const caption = getImageCaption(imgEl);
+    const parentEl = imgEl.closest('p');
+
+    if (!caption) {
+      const wrapper = createTag('div', null, imgEl.cloneNode(true));
+      row.append(wrapper);
+    } else {
+      const picture = createTag('p', null, imgEl.cloneNode(true));
+      const em = createTag('p', null, caption.cloneNode(true));
+      const wrapper = createTag('div', null);
```
Comment (meganthecoder) [inline scripts/utils.js:82]:
```suggestion
      const wrapper = createTag('div');
```

**Review by meganthecoder — APPROVED** [review]
I've approved the code, I don't really understand why you need to do this. Wouldn't using the bulk updater to make the content compatible with milo's figure be better than putting this code before LCP?

**Comment by rgclayton** [comment]
> > I've approved the code, but I don't really understand why you need to do this. Wouldn't using the bulk updater to make the content compatible with milo's figure be better than putting this code before LCP?
> 
> Hey @rgclayton @auniverseaway what do you think about rather using the bulk updater?

The blog auto converts images into a figure. I suppose we could look into using the figure block in place of the auto figure generation. I don't know the history here though. 

@auniverseaway Any thoughts?

**Comment by auniverseaway** [comment]
@saugatmalla @rgclayton I think @meganthecoder has a good point. You do pay for those bytes in pre-LCP, and since we're having to do a bulk migration anyway, maybe we just migrate the content and be done. You've now written the code to do that migration.

On the flip side, so much of what makes the blog, THE BLOG, is that it's not block heavy. If we start kicking out to blocks to solve everything, these pages will soon look like complicated product pages.

Either way works for me. Out of pure laziness, I would probably ship this as-is now, see where performance lands, and then re-evaluate if you need to eek out more perf.

**Pull request merged**